src/workerd/api/gpu/gpu-device.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include "gpu-bindgroup-layout.h"
+#include "gpu-buffer.h"
+#include <dawn/native/DawnNative.h>
+#include <webgpu/webgpu_cpp.h>
+#include <workerd/jsg/jsg.h>
+
+namespace workerd::api::gpu {
+
+class GPUDevice : public jsg::Object {
+  wgpu::Device device_;
+  jsg::Ref<GPUBuffer> createBuffer(jsg::Lock &, GPUBufferDescriptor);
+  jsg::Ref<GPUBindGroupLayout>
+  createBindGroupLayout(GPUBindGroupLayoutDescriptor descriptor);
```

Comment:
Our prevailing style is to put these into an explicit `private:` section after the `public:` section... so

```cpp
class GPUDevice : public jsg::Object {
public:
  explicit GPUDevice( ....

private:
  wgpu::Device device_;
  //...
```

Since this is still all work in progress, consider this non-blocking for now, but before this is finished it would be good for this to follow the idiomatic style.